modules/core/src/lib/layer.js

```diff
@@ -472,16 +472,16 @@ export default class Layer extends Component {
   getNumInstances(props) {
     props = props || this.props;
 
-    // First check if the layer has set its own value
-    if (this.state && this.state.numInstances !== undefined) {
-      return this.state.numInstances;
-    }
-
     // Check if app has provided an explicit value
     if (props.numInstances !== undefined) {
       return props.numInstances;
     }
 
+    // First check if the layer has set its own value
+    if (this.state && this.state.numInstances !== undefined) {
```

Comment:
Change is fine but as a side note: `numInstances` semantics feels rather shaky to me. I am not sure I can explain to a user exactly how this field is supposed to work. Might be good to brainstorm and change or at least update docs to cover current usage.